tests/test_common.py

```diff
@@ -4061,6 +4061,17 @@ class Ex(Struct, omit_defaults=True):
             sol = proto.encode(y)
             assert res == sol
 
+    def test_unset_encode_struct_array_like(self, proto):
+        class Ex(Struct, array_like=True):
+            x: Union[int, UnsetType] = UNSET
+            y: Union[int, UnsetType] = UNSET
+            z: int = 0
+
+        for x, y in [(Ex(), [0]), (Ex(x=1), [1, 0]), (Ex(y=2), [2, 0])]:
+            res = proto.encode(x)
+            sol = proto.encode(y)
+            assert res == sol
+
```

Comment:
We need another assertion here:

```python
roundtrip = proto.decode(res, type=Ex)
assert roundtrip == res
```

And add a `z` case for good measure.

What good is it to speak, if we will be misunderstood?